### DISCUSSION TITLE
Connection works from UI but not from cli

### DISCUSSION THREAD

**Discussion by biofilos** [GMOD/jbrowse-components#3735] [General]
Hello,

I am implementing a genome browser for researchers at our uni, and I would like to use connections.
When I set up a connection to the browser with
```
jbrowse add-connection -a hg38 -n ALFA -t UCSCTrackHubConnection --out html https://ftp.ncbi.nlm.nih.gov/snp/population_frequency/TrackHub/20200227123210/hub.txt
```
I get the config
```json
{
      "type": "UCSCTrackHubConnection",
      "connectionId": "UCSCTrackHubConnection-hg38-1685505389306",
      "name": "ALFA",
      "assemblyName": "hg38",
      "hubTxtLocation": {
        "uri": "https://ftp.ncbi.nlm.nih.gov/snp/population_frequency/TrackHub/20200227123210/hub.txt"
,
        "locationType": "UriLocation"
      }
    },
```
For now, I am working only with HG38. When I activate the connection in the browser, I get the error "Error: Cannot find assembly for "hg19" from the genomes file for connection "ALFA"".
However, if I go to "Add Connection" in the UI and add it with the same url to hub.txt, the connection works

As there are many track hubs that we want our users to have available, I want to set them up in advance. Is there a way to set them up with the cli so they are available whenever our users open the browser?

I saw [this discussion](https://github.com/GMOD/jbrowse-components/discussions/2755). I guess that would work in a pinch, though I also wonder if there is a better way to set up the connection via the cli so they work when only one assembly is installed

Thank you

**Comment by cmdcolin** [comment]
@biofilos which JBrowse version are you using? I just tested it on current main and it worked

![image](https://github.com/GMOD/jbrowse-components/assets/6511937/c9018422-1b8a-48f7-ad1b-cf7294f72b98)

**Reply by cmdcolin** [reply]
I used the CLI command that you suggested

**Reply by cmdcolin** [reply]
Actually my config included an hg19, so that may be why it worked for me, I will convert this to an issue so we can look into it

**Comment by cmdcolin** [comment]
I made a proposal to fix this in this PR https://github.com/GMOD/jbrowse-components/pull/3738

Alternatively, you can also fix it "today" in your config by changing "assemblyName":"hg38" to "assemblyNames":["hg38"]

**Reply by cmdcolin** [reply]
our CLI tool should be updated to properly output that updated config too

**Comment by biofilos** [comment]
Thanks a lot, that helps for now.

Just to clarify, When I see "TypeError: NetworkError when attempting to fetch resource", it is because the server of the track hub doesn't have allow CORS to my jbrowse locartion, or is there I can configure in my jbrowse instance to be able to display those tracks?

**Reply by cmdcolin** [reply]
not sure, CORS errors are hard to detect in javascript so jbrowse has a hard time detecting them and showing a proper error for them, but you can check the "network" tab in the chrome dev tools to see


 I tested it the hub you posted in my web browser and I don't see any CORS error, but hard to say, NCBI is sometimes weird about CORS stuff

**Comment by biofilos** [comment]
I see,
I went to the network tab, and with ALFA, I can see the ...chunk.js being pulled with 200, but I don't see anything in the browser. With the track hub Uniprot Features, I see the same 200 in the network tab, but also "TypeError: NetworkError when attempting to fetch resource" in the browser

More generally, is there any config I have to make to make publick track hubs work in general?
I tested with a little track hub I made in S3, and it works

Also, I would be happy to move this topic to another discussion if necessary
Thanks again!

**Reply by cmdcolin** [reply]
I don't think there is anything special to enable to make it work but there is a thing where on the web where the track hub itself has to support CORS. that is frustratingly not super common. one way this can be worked around is using a server side CORS proxy but it is recommended to self-host these https://httptoolkit.com/blog/cors-proxies/

note: jbrowse desktop does not have any CORS limitations, it can request data from resources that dont have CORS enabled, but anything in the web browser has to play by CORS rules

**Comment by biofilos** [comment]
oh, good to know!
thanks a lot for your help

cheers!
